File: rb/features.rb
# Geodescription SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module GeodescriptionFeatures
  def self.make_feature(name)
    case name
    when "base"
      GeodescriptionBaseFeature.new
    when "ratelimit"
      GeodescriptionRatelimitFeature.new
    when "retry"
      GeodescriptionRetryFeature.new
    when "test"
      GeodescriptionTestFeature.new
    when "timeout"
      GeodescriptionTimeoutFeature.new
    else
      GeodescriptionBaseFeature.new
    end
  end
end

File: py/features.py
# Geodescription SDK feature factory

from feature.base_feature import GeodescriptionBaseFeature
from feature.test_feature import GeodescriptionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GeodescriptionBaseFeature(),
        "test": lambda: GeodescriptionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
